pkg/config/kube/conversion.go

```diff
@@ -54,6 +54,20 @@ func ConvertProtocol(port int32, portName string, proto corev1.Protocol, appProt
 	name := portName
 	if appProto != nil {
 		name = *appProto
+		// Kubernetes has a few AppProtocol specific standard names defined in the Service spec
+		// Handle these only for AppProtocol (name cannot have these values, anyways).
+		switch name {
+		// "http2 over cleartext", which is also what our HTTP2 port is
+		case "k8s.io/http2":
+			return protocol.HTTP2
+		// "grpc traffic" - vague description, but the spec means gRPC over TLS.
+		// In Istio, GPRC means "gRPC over plaintext", so we actually use HTTPS.
+		case "k8s.io/grpc":
+			return protocol.HTTPS
+		//  plain tcp traffic
+		case "k8s.io/tcp":
```

Comment:
this protocol was removed from this iteration and added as a followup work to the KEP
https://github.com/kubernetes/enhancements/pull/3912